fix: attribute validation for generated script tag

kvn-shn opened the pull request:
### Before submitting the PR, please make sure you do the following
fixes https://github.com/sveltejs/kit/issues/1767

Strictly speaking the attributes `type`, `url` and `body` in the generated script tag are invalid. This PR replaces them with `data-type`, `data-url` and `data-body`.

### Tests
Adjusted the existing test cases.

### Changesets
Generated a minor changeset.

Conduitry commented:
This could potentially be a lot of data, so we're trying to avoid having the browser parse this as javascript. Are there any `type=` values that the validator won't complain about but which the browser won't parse and execute as javascript?

kvn-shn commented:
> This could potentially be a lot of data, so we're trying to avoid having the browser parse this as javascript. Are there any `type=` values that the validator won't complain about but which the browser won't parse and execute as javascript?

Okay, I understand. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-type) every MIME type other than JS should do the trick. I chose `text/plain` and adjusted the PR accordingly.

(Furthermore I noticed that there is also the invalid `body` attribute used. I renamed it to `data-body`  as well.)

benmccann commented on packages/kit/src/runtime/server/page/render.js:
```diff
@@ -159,9 +159,10 @@ export async function render_response({
-					return body
-						? `<script type="svelte-data" url="${url}" body="${hash(body)}">${json}</script>`
-						: `<script type="svelte-data" url="${url}">${json}</script>`;
+					const attrTag = 'data-type="svelte-data"';
```
you could just inline these first two variables. for the third I would try to keep the extra space from making it to the output even if it makes it a bit uglier

```
const attrBody = body ? ` data-body="${hash(body)}"` : '';
return `<script type="application/json" data-type="svelte-data" data-url="${url}"${attrBody}>${json}</script>`;
```

kvn-shn replied:
I tried to prioritize readability vs terseness. imho this is cleaner especially since Prettier would otherwise add a very awkward linebreak, see https://github.com/sveltejs/kit/blob/a604f8c2369055b75aabef0029b2d0fbeb16385a/packages/kit/src/runtime/server/page/render.js#L163-L165 But if you insist I'll do it the other way.

benmccann commented on .changeset/olive-maps-join.md:
```diff
@@ -0,0 +1,5 @@
+'@sveltejs/kit': minor
+---
+
+fix incorrect attributes in generated script tag
```
```suggestion
fix attribute validation in generated script tag
```

Rich-Harris commented:
thanks!

benmccann commented on .changeset/olive-maps-join.md:
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': minor
```
this should have been `patch`

kvn-shn replied:
thanks for the info. :) btw, this is not clear reading from the readme: https://github.com/sveltejs/kit#changelogs

benmccann replied:
Added!
